engine/src/main/java/org/terasology/rendering/world/dag/RenderNode.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.world.dag;
+
+import org.terasology.rendering.cameras.Camera;
+
+public interface RenderNode {
+
+    void initialise();
+
+    void preRender(float update);
+
+    void render();
+
+    void postRender();
```

Comment:
I'm not sure if this is needed now. I can see the aesthetics of having a pre/postRender method, but as I'm suggesting to rename preRender(), I wonder what would go in postRender() that can't go in process().

I guess we could have a preProcess() method executed before all nodes in the dag get processed and a postProcess() method executed after all nodes in the dag have been processed. At this stage I just don't know if we'd need them.